app/services/assignment_service.py

```diff
@@ -185,8 +201,15 @@ async def update_assignment(self, assignment: AssignmentModel, update_assignment
         if "due_date" in update_fields:
             assignment.due_date = update_fields["due_date"]
 
+        if "is_published" in update_fields:
+            if update_fields["is_published"] == False:
+                canvas_assignment = await LmsSyncService(self.session).get_assignment(assignment.id)
+                if canvas_assignment["unpublishable"] == False:
+                    raise AssignmentCannotBeUnpublished
+            assignment.is_published = update_fields["is_published"]
```

Comment:
In the professor extension, @frostyfan109 when a open/due date is changed, and we want to sync it up to Canvas - we will also have to update our local instance assignment status.

@Hoid you will need to update the `is_published` status for an assignment at line 201 as well. And also at line 198 when `available_date` is changed

Reply:
not sure i'm totally following this. the assignment status is computed, so after sync that status will recompute based on the status of published, available_date, due_date, current_time, etc.

Reply:
in the `update_assignment` method, when we update the available/due date of an assignment, we also need to update the is_published field to match do we not.

current implementation is not handling this?

Reply:
@ptlharit2 no, the `is_published` field is strictly a copy of the `published` field on assignments in Canvas, just in our DB. and like Griffin said the `AssignmentStatus` is always computed, not stored in the database. so we don't ever have to update that value, we only have to update the values it depends on, like `due_date` and `is_published`.